### PR TITLE
chore: dynamically creating mock files

### PR DESCRIPTION
**Pull request by Charles546**
#### Description
Instead of keeping the mockgen generated files in the repo, we should
run the mockgen whenever it is necessary. This way, we don't have to
worry about the linting of the generated files.

#### This PR fixes the following issues
N/A

### CODE REVIEW

**Review by wyardley — COMMENTED** [review]
@Charles546 can we add a Makefile, and add a dependency on the source files for the mocks they correspond to, both to keep the behavior consistent for local dev and in CI? That should simplify docs, and also (I think) let us have the rebuild happen when needed and reduce the chance that someone ends up having stale mocks?

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+	@go install ./...
+
+lint:
+	@echo -e "$(BOLD)Linting source code$(RESET)"
+	@golangci-lint run
+
+mockgen:
+	@echo -e "$(BOLD)Generating mocking code$(RESET)"
+	@go get github.com/golang/mock/mockgen@v1.4.4
+	@./run_mockgen.sh
```
Comment (wyardley) [inline Makefile:19]: does this really only ever have to be run once? Doesn't it have to be updated if the mocks need to be rebuilt? can we make this an actual (vs. phoney) make target on the actual files that need to be updated vs. their source files?

**Comment by Charles546** [comment]
@wyardley Added the files as dependencies in Makefile. Please review again.

**Review by wyardley — APPROVED** [review]

**Pull request merged**